### PR TITLE
Add docker.RunCommandUntil function to E2E framework

### LINKED ISSUE #767: Add docker.RunCommandUntil function to retry in tests
Current [docker.RunCommand](https://github.com/submariner-io/shipyard/blob/devel/test/e2e/framework/docker.go#L79) in [shipyard](https://github.com/submariner-io/shipyard)/[test](https://github.com/submariner-io/shipyard/tree/devel/test)/[e2e](https://github.com/submariner-io/shipyard/tree/devel/test/e2e)/[framework](https://github.com/submariner-io/shipyard/tree/devel/test/e2e/framework)/docker.go doesn't allow the command to fail, if it fails the test fails. Some tests require retrying docker command until the command succeeds, so docker.RunCommandUntil function needs to be added.

### PR DESCRIPTION
**Pull request by mkimuram**
This PR adds docker.RunCommandUntil function to E2E framework.

fixes: #767

### CODE REVIEW

**Comment by mkimuram** [comment]
@tpantelis @dfarrell07 @sridhargaddam  I tested with [this submariner-io/submariner code](https://github.com/mkimuram/submariner/commit/c3f861bec01c91466d73c47c88b0f14a8204ab1f) and confirmed that the test don't fail without fixed sleep. Please review this PR.

File: test/e2e/framework/docker.go [hunk 1]
```diff
@@ -76,7 +78,30 @@ func (d *Docker) RunCommand(command ...string) (string, string) {
...
+func (d *Docker) RunCommandUntil(command ...string) (string, string) {
+	var cmdErr error
+	var stdout, stderr string
+
+	err := wait.PollImmediate(5*time.Second, time.Duration(TestContext.OperationTimeout)*time.Second,
+		func() (bool, error) {
+			stdout, stderr, cmdErr = d.runCommand(command...)
+			if cmdErr != nil {
+				return false, nil // nolint:nilerr // Returning nil value is intentional
+			}
```
Comment (tpantelis) [inline test/e2e/framework/docker.go:102]: I think we still want to report the command error if the retry timeout is reached. You could use `framework.AwaitUntil` or keep track of the last error and report it below.

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: test/e2e/framework/docker.go (hunk 1 above)
Comment (mkimuram) [reply test/e2e/framework/docker.go:102]: @tpantelis  Thank you for your comments. I've changed it to report the command errors. This aligns to the behavior of [`IsTransientError`](https://github.com/submariner-io/shipyard/blob/2982bca7e57d74891fe9dcdee876e48fcaa1c3ac/test/e2e/framework/api_errors.go#L27-L34) called from  [`AwaitResultOrError`](https://github.com/submariner-io/shipyard/blob/devel/test/e2e/framework/framework.go#L524). Now it outputs errors like below and keep retrying:

```
STEP: Sending an http request from external app "172.117.0.4" to "10.1.192.6" in the cluster "cluster1"
Mar 31 14:42:03.945: INFO: Transient failure when attempting to run [curl -m 3 10.1.192.6:80/e2e-tests-ext-dataplane-kdg9fcluster1]: exit status 28
```

Comment (tpantelis) [reply test/e2e/framework/docker.go:102]: OK. My only nit is that it's not known if it's a transient error so perhaps change the output to "_Error attempting to run..._"

Comment (mkimuram) [reply test/e2e/framework/docker.go:102]: I agree. Fixed and squashed the commits.

**Review by dfarrell07 — APPROVED** [review]

**Review by tpantelis — APPROVED** [review]

**Pull request merged**